### DISCUSSION TITLE
Can you disable html overflow="hidden" when Dialog is open?

### DISCUSSION THREAD

**Discussion by simula-r** [tailwindlabs/headlessui#2181] [Help]
Opening a dialog sets the HTML element inline style as "overflow: hidden" 
I am assuming this is to prevent scrolling, however:
This is causing elements with position: sticky to disappear. 

https://user-images.githubusercontent.com/18093452/212228895-0b87dd0b-a025-4698-8c07-0ce8f69b2d6c.mov

**Comment by christian-reichart** [comment]
same problem, making this optional would be great.

Or maybe change this behaviour to overflow: clip, which is relatively new though: https://caniuse.com/?search=overflow%3A%20clip

**Comment by EricDotSmith** [comment]
I'm also having a similar issue here: https://github.com/tailwindlabs/headlessui/discussions/2305

Did you figure out a work around for this problem?

**Reply by christian-reichart** [reply]
I switched to the Dialog of Reach UI

**Reply by EricDotSmith** [reply]
I just tried playing around with it and while it's really great it doesn't quite have the feel I want from a modal like the HeadlessUI Dialog does. I'm thinking of just keeping it and accepting the flicker for now, albeit I hope the team or someone else from the community can figure out why this is happening and fix it.

I'm using this almost line for line: https://headlessui.com/react/dialog

**Reply by christian-reichart** [reply]
If you want a predefined component you could look broader, the main benefit of HeadlessUI and Reach is that you can customize everything. If you don't need it, there are probably many UI kits out there that do the job.

**Comment by EricDotSmith** [comment]
Also I can confirm the issue is the overflow: hidden. So far I can't find a good way to override this while maintaining the same behaviour 🤷‍♂️

**Comment by psugihara** [comment]
Anyone figure out a fix for this? Hitting the same thing with a sticky header.


https://user-images.githubusercontent.com/282016/233129286-d1a3e186-3dd7-48be-8f40-2d5e44c51921.MP4

**Reply by EricDotSmith** [reply]
I would love to know an answer to this as well. But in the meantime I'm considering switching my modals to use preline. Idk if their modals have this issue or not but we'll see.

**Comment by sean-clayton** [comment]
Having this _exact_ issue with my project: https://github.com/13vaults/13vaults.com/blob/218f092d851c2e5f2f23114c7153c62438db7dbe/components/mega-nav.tsx (https://www.13vaults.com). I only just now realized it was a bug with the dialog component + a sticky element. Losing my hair for weeks over this :sweat_smile: 

Would love to see a way to address this--it might just be a thing only a webkit update can fix though :crying_cat_face:

**Reply by sean-clayton** [reply]
`@reach/dialog` works just fine for my use-case. See the diff here how I refactored my component to use `@react/dialog` and `framer-motion` instead of headless-ui's dialog and transition components: https://github.com/13vaults/13vaults.com/commit/d13f4717af79e1216c96073bdcc7b2526e6937f6#diff-a75fd728c03405a0d9e680b949580052ab2c5b741a579b1f9daa71dc430d1f3c

**Comment by hodenortypro** [comment]
Yes, it is possible to disable the overflow="hidden" attribute of the HTML element when a dialog is open. This can be done using JavaScript/jQuery by adding a CSS class to the HTML element when the dialog is open.

Here's an example of how this can be achieved using jQuery:

javascript
Copy code
// Open dialog function
function openDialog() {
   $('body').addClass('dialog-open');
   // code to open the dialog
}

// Close dialog function
function closeDialog() {
   $('body').removeClass('dialog-open');
   // code to close the dialog
}
In the above code, the openDialog() function adds a CSS class called dialog-open to the body element when the dialog is opened. This class can be used to override the overflow="hidden" attribute of the HTML element in CSS, allowing the page to scroll while the dialog is open.

Here's an example CSS code that would disable overflow="hidden" when the dialog-open class is present:

css
Copy code
.dialog-open {
   overflow: visible !important;
}
With this code in place, the overflow="hidden" attribute will be disabled when the dialog is open and re-enabled when the dialog is closed.



[ballsportsgames](https://www.ballsportsgames.com/)

**Reply by tourlida** [reply]
where we should at this class?

**Reply by sean-clayton** [reply]
It looks a lot like a chat-gpt generated response.

**Reply by robinsimonklein** [reply]
Yeah, jQuery has nothing to do here...

**Comment by benjaminarteaga** [comment]
Same issue here... Please provide a solution.

**Comment by juanmanuel-so** [comment]
Same issue :(

**Comment by v-dev-cl** [comment]
Same issue with the current version 1.7.15.

**Reply by v-dev-cl** [reply]
In my case was my own fault, I had set the element in a sub element causing the issue, so just moving the element one level upper fixed my issue.

**Comment by ITSFerroli** [comment]
I solve this issue forcing to use always the css property that i want, like so:

`html {
    overflow: clip !important;
}`

so the inline overflow hidden style is ignored.

**Reply by lalcaraz1** [reply]
This fixed my issue, thanks a lot!

**Reply by christian-reichart** [reply]
If you need to support Safari, overflow: clip might not be a good solution yet, it's not really supported sadly.

**Comment by jessems** [comment]
I got around this with a bit of a hack:

```js
  useEffect(() => {
    let timer: number;

    if (open) {
      timer = setTimeout(() => {
        const html = document.documentElement;
        html.style.overflow = 'auto'; // revert to default overflow
      });
    }

    return () => {
      clearTimeout(timer);
    };
  }, [open]);
```

Wrapping it in a setTimeout puts it in a different execution cycle (I guess). You don't even need to introduce an actual delay.

**Reply by alaminShaheen** [reply]
doing this makes the body scrollable even when the dialog is open.

**Comment by dada878** [comment]
I have a sticky sidebar, and when a Headless UI modal opens, the sidebar will not stay sticky.
The following code solved the issue in my case.

```css
html {
  overflow: visible !important;
}
```
or using tailwind
```html
 <html className="!overflow-visible">...<html>
```

**Comment by alex-carter-mobdev** [comment]
html:has(dialog[open]) {
  /* remove the main scrollbar when dialog is open */
  overflow: hidden;
}
 this will help

**Comment by saimonkat** [comment]
Fix for this issue, which you can use in `global.css`/`main.css`

```
@layer base {
  html {
    @apply !overflow-visible !p-0;
  }
}
```

**Comment by aminkhadivar** [comment]
I fix this issue,  you must add styled in custom.css .
```javascript

body {
    overflow: unset !important;
}
```

**Comment by n3ps** [comment]
Can someone explain the reasoning behind Headless UI adding padding and overflow by default?

**Comment by KevinDevNull** [comment]
I think, apply `overflow-x: clip;` in body should solve the problem.

**Comment by LosGlennos** [comment]
I don't have this issue with Dialog, but I've had this issue with Popover and Listbox. Writing this to help people who has this problem with other types of components as well. These components has a `modal` property which you can set to false. This helps with keeping the stickiness, but it's not optimal as focus disappears and you can scroll what's behind the open component.

Example

```
<PopoverPanel
    as="ul"
    portal
    anchor="bottom end"
    **modal={false}** >
```

As stated, it's not a perfect solution.. have anyone found anything better yet?

**Comment by emadha** [comment]
use `modal={false}` in MenuItems
```
<MenuItems modal={false}>
```

**Comment by tarynficklin** [comment]
I wish with every fiber of my being that the devs would make this an optional prop on the modal and other dropdown menus. Please devs, we beg of you!
